src/Shmuelie.WinRTServer/Internal/BaseClassFactoryWrapper.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Shmuelie.WinRTServer.Windows.Com;
+using Shmuelie.WinRTServer.Windows.Com.Marshalling;
+using IUnknown = Windows.Win32.System.Com.IUnknown;
+using static Windows.Win32.PInvoke;
+
+namespace Shmuelie.WinRTServer.Internal;
+
+[GeneratedComClass]
+internal partial class BaseClassFactoryWrapper(BaseClassFactory factory, ComWrappers comWrappers) : IClassFactory
+{
+    [return: MarshalUsing(typeof(HResultMarshaller))]
+    public unsafe global::Windows.Win32.Foundation.HRESULT CreateInstance(void* pUnkOuter, Guid* riid, void** ppvObject)
+    {
+        if (pUnkOuter is not null)
+        {
+            return global::Windows.Win32.Foundation.HRESULT.CLASS_E_NOAGGREGATION;
+        }
+
+        if (!riid->Equals(IUnknown.IID_Guid) && !riid->Equals(factory.Iid))
+        {
+            return global::Windows.Win32.Foundation.HRESULT.E_NOINTERFACE;
+        }
+
+        try
+        {
+            var instance = factory.CreateInstance();
+            var unknown = comWrappers.GetOrCreateComInterfaceForObject(instance, CreateComInterfaceFlags.None);
```

Comment:
You're never releasing this

Reply:
If I do that though the object will instantly be at zero before I return?

Reply:
This is correct if the case on line 34 is taken.

The branch for line 38 does another QI, which means now you are up to two. Additionally, if 39 is true, `unknown` is leaking. If it is false, you've QI'd twice but are only transfering ownership of a single `AddRef` to the caller, line 30 or line 38. I would call `Marshal.Release()` on `unknown` immediately after line 38. Regardless of the result, you want to release `unknown` at that point.